### PR TITLE
[0.9.1 save update] Add days passed to Campaign SaveData and fix unsaved CampaignID

### PR DESCRIPTION
**Pull request by eos428**
There's no issue written for either of these yet, so this wouldn't close any of our current issues.

Important note: Save version is increased again

### CODE REVIEW

File: src/fheroes2/system/settings.h [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include "players.h"
 #include "system.h"
 
+#define FORMAT_VERSION_091_RELEASE 9100
```
Comment (idshibanov) [inline src/fheroes2/system/settings.h:36]: There are two additional savegame changing PRs coming in. I would avoid going straight to 9100 for now. Have to decide the strategy around it.

Comment (eos428) [reply src/fheroes2/system/settings.h:36]: In that case, why not have all 3 PRs increase the version to the same number? I think it wouldn't make sense if we need to increase version for every one of the PRs, especially if let's say, they'd be merged within hours/days of each other.

Comment (idshibanov) [reply src/fheroes2/system/settings.h:36]: Could work. @ihhub what do you think?

Comment (ihhub) [reply src/fheroes2/system/settings.h:36]: Yes. We can do this. I can fix merge conflicts as well. Which is the third PR besides #2630?

File: src/fheroes2/campaign/campaign_savedata.h [hunk 2]
```diff
@@ -82,6 +88,7 @@ namespace Campaign
         std::vector<std::string> _earnedCampaignAwards; // should have its own data format
         int _currentScenarioID;
         int _campaignID;
+        int _daysPassed;
```
Comment (ihhub) [inline src/fheroes2/campaign/campaign_savedata.h:91]: The number of days cannot be negative value plus `world.CountDay()` returns `uint32_t`. Could you please fix it?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @eos428 , everything looks good except one place. Could you please fix it?

File: src/fheroes2/campaign/campaign_savedata.h [hunk 3]
```diff
@@ -65,12 +65,18 @@ namespace Campaign
             return _finishedMaps.empty();
         }
 
+        int getDaysPassed() const
```
Comment (ihhub) [inline src/fheroes2/campaign/campaign_savedata.h:68]: Please return here `uint32_t`.

File: src/fheroes2/campaign/campaign_savedata.h [hunk 4]
```diff
@@ -65,12 +65,18 @@ namespace Campaign
...
+            return _daysPassed;
+        }
+
         void setCurrentScenarioBonus( const ScenarioBonusData & bonus );
         void setCurrentScenarioID( const int scenarioID );
         void setCampaignID( const int campaignID );
         void addCurrentMapToFinished();
         void addCampaignAward( const std::string & award );
         void reset();
+        void addDaysPassed( const int days );
```
Comment (ihhub) [inline src/fheroes2/campaign/campaign_savedata.h:79]: Please change it to `uint32_t`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @eos428 , could you please change int to uint32_t in highlighted places? The rest is perfect.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@eos428 , thank you veru much for this fix!

**Pull request merged**